redash/tasks/queries.py

```diff
@@ -425,7 +435,13 @@ def run(self):
         if error:
             self.tracker.update(state='failed')
             result = QueryExecutionError(error)
+            if self.scheduled_query:
+                self.scheduled_query.schedule_failures += 1
+                models.db.session.add(self.scheduled_query)
         else:
+            if self.scheduled_query:
+                self.scheduled_query.schedule_failures = 0
```

Comment:
How about we do this update only if `schedule_failures` is `> 0`? Otherwise we generate unneeded update.